### ISSUE TITLE
Using pg_search_scope with ignoring accents on thread raise an exception

### ISSUE THREAD

**Issue by thiaguerd** [Casecommons/pg_search#446]
Hello!

I updated to latest version of gem and I give a problem let's see

```ruby
class User < ApplicationRecord
  pg_search_scope :search_by_name,
                  against: :name,
                  ignoring: :accents,
                  using: {
                    tsearch: {
                      prefix: true,
                      negation: true
                    }
                  }
end
```

Normal use no problem

```ruby
User.search_by_name('a').count
```

But if is in thread raise an exception

```ruby
Thread.new { User.search_by_name('a').count }
```

Message is

```text
arel/collectors/plain_string.rb:15:in `<<': no implicit conversion of nil into String (TypeError)
```

Full message is

```text
Traceback (most recent call last):
        21: from (irb):2:in `block in irb_binding'
        20: from pg_search/lib/pg_search/model.rb:20:in `block in pg_search_scope'
        19: from pg_search/lib/pg_search/scope_options.rb:20:in `apply'
        18: from pg_search/lib/pg_search/scope_options.rb:148:in `rank_join'
        17: from pg_search/lib/pg_search/scope_options.rb:88:in `subquery'
        16: from pg_search/lib/pg_search/scope_options.rb:142:in `rank'
        15: from pg_search/lib/pg_search/scope_options.rb:142:in `gsub'
        14: from pg_search/lib/pg_search/scope_options.rb:143:in `block in rank'
        13: from pg_search/lib/pg_search/features/tsearch.rb:20:in `rank'
        12: from pg_search/lib/pg_search/features/tsearch.rb:173:in `tsearch_rank'
        11: from pg_search/lib/pg_search/features/tsearch.rb:141:in `tsdocument'
        10: from pg_search/lib/pg_search/features/tsearch.rb:141:in `map'
         9: from pg_search/lib/pg_search/features/tsearch.rb:142:in `block in tsdocument'
         8: from pg_search/lib/pg_search/features/tsearch.rb:198:in `column_to_tsvector'
         7: from pg_search/lib/pg_search/features/feature.rb:42:in `normalize'
         6: from pg_search/lib/pg_search/normalizer.rb:22:in `add_normalization'
         5: from gems/activerecord-6.0.3.3/lib/arel/nodes/node.rb:38:in `to_sql'
         4: from gems/activerecord-6.0.3.3/lib/active_record/connection_adapters/determine_if_preparable_visitor.rb:10:in `accept'
         3: from gems/activerecord-6.0.3.3/lib/arel/visitors/visitor.rb:11:in `accept'
         2: from gems/activerecord-6.0.3.3/lib/arel/visitors/visitor.rb:30:in `visit'
         1: from gems/activerecord-6.0.3.3/lib/arel/visitors/to_sql.rb:345:in `visit_Arel_Nodes_NamedFunction'
gems/activerecord-6.0.3.3/lib/arel/collectors/plain_string.rb:15:in `<<': no implicit conversion of nil into String (TypeError)
```

Because in threads PgSearch.unaccent_function is nil 

Try this

```ruby
Thread.new { puts PgSearch.unaccent_function }
```

**To fix this in my project I just make this**

```ruby
module PgSearch
  mattr_accessor :unaccent_function
  self.unaccent_function = 'unaccent'
end
```

I see rubocop warning  ```Avoid mutating class and module attributes. (convention:ThreadSafety/ClassAndModuleAttributes)``` but how set a default value and avoid nil when called in threads?

I think that asap in active support we have a ```thread_mattr_accessor``` with a default value, see [here](https://github.com/rails/rails/blob/master/activesupport/lib/active_support/core_ext/module/attribute_accessors_per_thread.rb#L143)

What can we do about it?

**Comment by fabioxgn** [comment]
I have the same problem. I tried to reproduce it in a feature test using Capybara but it just breaks in production/development when running with Puma.

**Comment by alispat** [comment]
I got the same issue here with unaccent...

**Comment by connorshea** [comment]
The best guess I've got is that this commit broke it? https://github.com/Casecommons/pg_search/commit/7ff41966026d990db95cda8ad4414f33264e734d

**Comment by connorshea** [comment]
I just reverted that commit in [my fork](https://github.com/connorshea/pg_search), then pointed my Gemfile at that fork and it works correctly. So that's definitely the commit that broke things.

And FWIW I don't use unaccent anywhere, just `multisearch_options` and it also effects me.

@nertzy I'd recommend reverting that commit and pushing out a new release since this bug can easily get past tests (where puma isn't used) and then cause problems when it actually gets to production. No rush, this is open source ofc, just want to make sure you're aware :) Thanks for pg_search!

**Comment by neontuna** [comment]
Just wanted to comment to maybe help others searching for this issue, this also appears to be causing `no implicit conversion of nil into Hash` but only in production, similar to the other issues.  Using PgMultisearch and the following options

```
PgSearch.multisearch_options = {
   using: {
    tsearch: {
      dictionary: 'english',
      tsvector_column: 'tsv_content',
      prefix: true
    }
  }
}
```

**Comment by aried3r** [comment]
@connorshea is right I believe. I created a repo with a [SSCCE](http://www.sscce.org/).

```console
$ git clone https://github.com/aried3r/rails_pg_search_multisearch_bug
$ cd rails_pg_search_multisearch_bug
$ bin/setup
$ rails server
```

Then open a browser and go to http://localhost:3000/search or `curl http://localhost:3000/search`.

**Comment by aried3r** [comment]
I have opened a PR at #447. Thanks @connorshea for the research!

I was wondering, @fabioxgn, have you found a way to write a failing test yet? I have been unable to, so far. Even with puma used in specs.

**Comment by fabioxgn** [comment]
> I have opened a PR at #447. Thanks @connorshea for the research!
> 
> I was wondering, @fabioxgn, have you found a way to write a failing test yet? I have been unable to, so far. Even with puma used in specs.

No. I tried to write a feature test using capybara but it works just fine in the test. Maybe because a single thread is used? 🤔

**Comment by turizoft** [comment]
I'm having the same issue, and can confirm what @thiaguerd and @fabioxgn say. It only fails when invoked within a thread. In this case Rails console vs Puma. I can provide more context if it helps somehow.

`Shop.first.products.published.with_attached_main_image.search_by_name("fig").pluck(:name)`
`=> ["Figura de Adam Jensen (Deus Ex)", "Han Solo Funko POP! Figure"]`
Works on Rails console.

But `products = @current_shop.products.published.with_attached_main_image.search_by_name(query)` in controller will fail.

config in model:

```
pg_search_scope :search_by_name,
                against: :name,
                using: { trigram: { threshold: 0.1 } },
                ignoring: :accents
```

**Comment by nertzy** [comment]
Fixed in [v2.3.4](https://github.com/Casecommons/pg_search/releases/tag/v2.3.4)

**Comment by connorshea** [comment]
Thanks!
